languages/AccountBlocks/en.ini

```diff
@@ -0,0 +1,9 @@
+Block Check Out = "You can't checkout more items"
+Block Electronic Resource Access = "You can't access electronic resources"
+Block Hold = "You can't place a hold"
+Block Recall = "You can't make a recall"
+Block Renewal = "You can't renew checkouts"
+Block Request Item = "You can't make a request on library material"
+Trap For Lost Card = "You card is reported as lost"
+Trap For Message = "You have a message from your library"
+Trap For Pickup = "Your requested material is available for pickup"
```

Comment:
Should these "Trap" messages really be blocking users from performing actions in the ILS? Some of these seem more like informational notes rather than reasons to restrict behavior. Have you tested that the system behaves appropriately in these cases?

That aside, though, I do like the idea of having a text domain for block messages. I wonder if there is existing functionality in other drivers that could be improved to use this.

Reply:
"Trap For Message" and "Trap For Pickup" are defined as "Notify the User that ...." But "Trap For Lost Card" is defined as "Do not allow any transactions using this user card to proceed." But it should not block user ta make actions using on-line account...

So theoretically, all traps are more like notifies then blocks. I tweaked the blocks handling a bit.

I do think some drivers could benefit from AccountBlocks text domain. It depends if the block description is free text in given system or something predefined - if it is predefined it makes sense to try to translate the value using this new text domain.

I do not know all systems VuFind has driver for.

The getAccounBlocks method is defined in these driver: Alma, Demo, Koha, KohaILSDI, KohaRest, SierraRest and VoyagerRestful.

Voyager looks like it has some predefined block reasons, so it could be a candidate. Also SierraRest looks like it uses some kind of block reason codes instead of free text. KohaRest looks like a candidate too.

Older Koha and KohaILSDI drivers uses only text comment from the db, so they could not use translation.

Reply:
Thanks for the improvements -- this does seem better, and it also makes it easy to subclass and change behavior if local needs differ from defaults.

Regarding the AccountBlocks text domain, I agree that most existing drivers don't look like they would fit with that approach since they're using free-form text pulled from the ILS. It looks to me like the KohaRest driver is the strongest candidate for using this text domain, which might be possible if we just move some strings from the main language files into the new text domain. I'd be interested to hear what @EreMaijala thinks of that idea; I want to be sure we're all on the same page about this approach before merging this PR, but I think once we have discussed that last detail, this is ready to go.

Reply:
Superficially the text domain makes sense, but there are several things to consider:

1. I'd like to avoid terminology that's specific to a certain interface and aim for more generic block reasons. And when the translation text is really a code that's not displayable as is, I'd rather use keys that look more like codes, such as 'request_block' or 'message_from_library'.
2. When the text is not a block, AccountBlocks is not the right domain name. ILSMessages or something maybe?
3. I think we should take a broader look on strings coming from ILS drivers and how they're translated. Particularly when connecting to multiple ILS's with MultiBackend, there's the need to translate strings independently per ILS. That's where Alma driver's translation prefix setting is handy, as you can define a per-library prefix or text domain. And the KohaRest driver should support the same mechanism. Things would be more clear if we had a well-defined list of common translation keys that the drivers should use for block statuses etc. that probably don't require library-specific overrides (compared to something like location codes which definitely do).

Reply:
Thanks for your feedback @EreMaijala 

ad 1) If we could define a set of block reasons that will work with the variety of systems, it would be probably best solution

ad 2) I am not against adding another domain and make it all semantically more precise.. but maybe then would be good idea to add a new method to drivers and reflect it in flash messages that way (maybe not styled as warning, but as info)

ad 3) I agree, that there is room for improvement I am just not sure how both ways would work together... But if we combine these two approaches we must be able to say clearly which strings  should be translated by default/common domain and which could be translated with prefix/domain configured per ILS driver

Reply:
For now I will try to do something about the 1)

Reply:
Thanks, @xmorave2 -- I agree that as a starting point, it makes sense to use more code-like keys in the translation file and to rename the text domain to ILSMessages (or something similar) to allow us to incorporate a broader range of ILS-related text in the future. As @EreMaijala points out, it would be really useful to group these messages together so that future driver developers can more easily find and reuse them. Regarding point 3, I think the solution could be as simple as having a configurable text domain in XCNCIP2.ini which defaults to ILSMessages -- then it's easy to create a custom text domain for per-driver messages, but by default, everything just works "normally."

There's clearly a lot more work that we could do here, but I'll be happy if we just set the stage effectively here, and we can do further work in subsequent PRs.

Reply:
I moved away from direct messages to codes and made translation domain configurable defaulting to 'ILSMessages'